internal/db/store.go

```diff
@@ -110,3 +110,24 @@ func (q *Queries) GetRuleEvaluationByProfileIdAndRuleType(
 	return ListRuleEvaluationsByProfileIdRow{},
 		fmt.Errorf("GetRuleEvaluationByProfileIdAndRuleType - expected 1 row, got %d", len(res))
 }
+
+// WithTransaction wraps an operation in a new DB transaction.
+// Ideally this would be a method of the Store interface, but Go's generics do
+// not allow for generic methods :(
+func WithTransaction[T any](store Store, fn func(querier ExtendQuerier) (T, error)) (result T, err error) {
```

Comment:
Not your fault, but I wish we had a better name for the type we return from GetQuerierWithTransaction than ExtendQuerier..

Reply:
the name is funky, but remember you don't need to pass `ExtendedQuerier` though, we could pass `Store` as well which is a more general interface.